app/models/project_step.rb

```diff
@@ -225,22 +242,22 @@ def finalize
   end
 
   # Returns number of days that the step's calendar date is about to be shifted.
-  # - If step is incomplete, returns number of days the scheduled date has been shifted.
+  # - If step is incomplete, returns number of days the scheduled start date has been shifted.
   # - If step is about to be set as complete, returns number of days late or early (negative).
   # - If step was already complete, returns number of days completed date has been shifted.
   # Assumes that record has pending changes assigned, but not yet saved.
   def pending_days_shifted
     return 0 unless is_finalized?
 
     # If completed date just got set.
-    if scheduled_date && completed_date && completed_date_changed? &&
-      completed_date_was.blank? && completed_date > scheduled_date
-      return (completed_date - scheduled_date).to_i
+    if scheduled_start_date && completed_date && completed_date_changed? &&
+      completed_date_was.blank? && completed_date > scheduled_start_date
+      return (completed_date - scheduled_start_date).to_i
     end
 
     # If incomplete and scheduled date changed.
-    if !completed? && scheduled_date_changed? && scheduled_date_was && scheduled_date
-      return (scheduled_date - scheduled_date_was).to_i
+    if !completed? && scheduled_start_date? && scheduled_start_date && scheduled_start_date
```

Comment:
Should this be `scheduled_start_date_changed?` instead of `scheduled_start_date?`?